BDArmory/Weapons/Missiles/MissileLauncher.cs

```diff
@@ -1319,12 +1319,11 @@ public void MissileLaunch()
                 part.explosionPotential = 0; // Minimise the default part explosion FX that sometimes gets offset from the main explosion.
                 rcsClearanceState = (GuidanceMode == GuidanceModes.Orbital && hasRCS && vacuumSteerable && (vessel.InVacuum()) ? RCSClearanceStates.Clearing : RCSClearanceStates.Cleared); // Set up clearance check if missile hasRCS, is vacuumSteerable, and is in space
 
-                var mml = part.GetComponent<MultiMissileLauncher>();
-                var isClusterMissile = mml && mml.isClusterMissile;
+                bool isClusterMissile = multiLauncher && multiLauncher.isClusterMissile;
                 if (!string.IsNullOrEmpty(exhaustPrefabPath))
                 {
-                    HashSet<Transform> dummyTransforms = isClusterMissile ? part.GetComponentsInChildren<MissileDummy>().SelectMany(md => md.transform.parent.GetComponentsInChildren<Transform>().Where(t => t.name == "exhaustTransform")).ToHashSet() : [];
-                    foreach (var t in part.FindModelTransforms("exhaustTransform"))
+                    HashSet<Transform> dummyTransforms = isClusterMissile ? part.GetComponentsInChildren<MissileDummy>().SelectMany(md => md.transform.parent.GetComponentsInChildren<Transform>().Where(t => t.name == multiLauncher.exhaustTransformName)).ToHashSet() : [];
```

Comment:
The `t.name == multiLauncher.exhaustTransformName` should be `t.name == boostTransformName` as it's finding those in the submunitions that match the same as in the parent missile in order to exclude them.

Similarly for `t.name == boostExhaustTransformName` below.

This also makes the changes to MultiMissileLauncher.cs unnecessary.